staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -633,6 +633,10 @@ func (a *Authenticator) AuthenticateToken(ctx context.Context, token string) (*a
 		return nil, false, err
 	}
 
+	if len(username) == 0 {
```

Comment:
should we move this into the `getUsername` function and limit it to the celMapper branch, and continue allowing present-but-empty jwt `sub` claims?

Requiring new cel mapping code to return a non-empty value sounds good to me (and that's where we're more likely to get a bug that returned an empty string, I think).

Before we tighten existing jwt behavior, I'd want to know how to check if tokens like this were being encountered

Reply:
Okay so I have left #121329 open for now, and just tightened the check on the `celMapper` branch.